src/en/mangahot/src/eu/kanade/tachiyomi/extension/en/mangahot/MangaHot.kt

```diff
@@ -0,0 +1,310 @@
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.decodeFromStream
+import kotlinx.serialization.json.put
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import okio.Buffer
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.injectLazy
+import java.net.URLEncoder
+import kotlin.math.ceil
+
+class MangaHot : HttpSource() {
+
+    override val name = "MangaHot"
+
+    override val baseUrl = "https://mangahot.to"
+
+    override val lang = "en"
+
+    override val supportsLatest = false
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimit(2)
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder().apply {
+        add("Referer", "$baseUrl/")
+    }
+
+    private val apiHeaders = headersBuilder().apply {
+        add("Accept", "*/*")
+        set("Referer", "$baseUrl/list")
+        add("Sec-Fetch-Dest", "empty")
+        add("Sec-Fetch-Mode", "cors")
+        add("Sec-Fetch-Site", "same-origin")
+    }.build()
+
+    private val json: Json by injectLazy()
+
+    private var currentTotalNumberOfPages = 1
+
+    // ============================== Popular ===============================
+
+    override fun popularMangaRequest(page: Int): Request =
+        GET("$baseUrl/api/list/latest?page=$page", apiHeaders)
+
+    override fun popularMangaParse(response: Response): MangasPage =
+        searchMangaParse(response)
+
+    // =============================== Latest ===============================
+
+    override fun latestUpdatesRequest(page: Int): Request =
+        throw UnsupportedOperationException()
+
+    override fun latestUpdatesParse(response: Response): MangasPage =
+        throw UnsupportedOperationException()
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val apiHeadersBuilder = apiHeaders.newBuilder()
+        val url = "$baseUrl/api".toHttpUrl().newBuilder()
+        val filterList = if (filters.isEmpty()) getFilterList() else filters
+        val tag = (filterList.find { it is TagFilter } as TagFilter).toUriPart()
+
+        if (page == 1) {
+            setTotalNumberPages(query, tag)
+        }
+
+        return when {
+            query.isNotBlank() -> {
+                url.addPathSegments("search")
+                val payload = buildJsonObject {
+                    put("keyword", query)
+                    put("page", page)
+                    put("size", PAGE_LIMIT)
+                }
+                val body = payload.toString().toRequestBody(JSON_MEDIA_TYPE)
```

Comment:
```suggestion
                val body = json.encodeToString(payload).toRequestBody(JSON_MEDIA_TYPE)
```
also import encodeToString